Dockerfile

```diff
@@ -1,23 +1,38 @@
-FROM alpine:3.15 AS download
+FROM alpine:3.16 AS download
 
-ENV PGTAP_VERSION=1.2.0
+ENV PGTAP_VERSION=1.2.1
 
-RUN apk add curl \
+# RUN apk add curl \
+#     && mkdir -p /opt && cd /opt \
+#     && curl -sq -LO http://api.pgxn.org/dist/pgtap/$PGTAP_VERSION/pgtap-$PGTAP_VERSION.zip \
+#     && unzip pgtap-$PGTAP_VERSION.zip \
+#     && apk del curl \
+#     && rm -rf /var/cache/apk/* /tmp/*
+
+RUN apk add git \
     && mkdir -p /opt && cd /opt \
-    && curl -sq -LO http://api.pgxn.org/dist/pgtap/$PGTAP_VERSION/pgtap-$PGTAP_VERSION.zip \
-    && unzip pgtap-$PGTAP_VERSION.zip \
-    && apk del curl \
+    && git clone https://github.com/theory/pgtap.git pgtap-$PGTAP_VERSION \
+    && cd /opt/pgtap-$PGTAP_VERSION \
+    && apk del git \
     && rm -rf /var/cache/apk/* /tmp/*
 
-# RUN apk add git \
-#     && mkdir -p /opt && cd /opt \
-#     && git clone https://github.com/theory/pgtap.git pgtap-$PGTAP_VERSION \
-#     && apk del git \
-#     && rm -rf /var/cache/apk/* /tmp/*
+FROM postgres:15beta3-alpine3.16 AS build-pgtap-psql-15
 
-FROM postgres:14-alpine AS build-pgtap-psql-14
+ENV PGTAP_VERSION=1.2.1
```

Comment:
I got curious whether repeating this line could be avoided for each build stage, but it's a little messy. If it was just a ARG instead of an ENV, it'd be easier, but there are a few suggestions here: https://stackoverflow.com/questions/52904847/how-do-i-copy-variables-between-stages-of-multi-stage-docker-build

Reply:
I don’t want it to be an ARG (can be overridden from outside), but exporting a `VERSIONS` file might solve this. Probably not worth it for this release. Can’t use the base approach, because we’re explicitly building in *separate* environments specifically to get different versions of PostgreSQL.